app/Services/Media/PhotoSaverService.php

```
@@ -143,11 +144,24 @@ public static function saveImageS3(
+            $maxRetries = 3;
+            $retryDelay = 1; // seconds
+
+            for ($retry = 0; $retry <= $maxRetries; $retry++) {
+                try {
+                    Storage::disk($disk)->put(
+                        $projectRef . '/' . $fileName,
+                        $imageContent
+                    );
+                    break; // Success, exit retry loop
+                } catch (Throwable $e) {
+                    if ($retry === $maxRetries || !($e instanceof S3Exception && Common::isRetryableError($e))) {
```

PHPMD (2.15.0)

Avoid using static access to class '\ec5\Libraries\Utilities\Common' in method 'saveImageS3'. (Clean Code Rules)

(StaticAccess)